ws/DiscordWebsocket.ts

```diff
@@ -184,6 +189,12 @@ export default class DiscordWebsocket {
         this.reset().connect();
         break;
       }
+      case GATEWAY_CLOSE_EVENT_CODES.RECONNECT: {
```

Comment:
Can we handle reconnect here? Remove the `.disconnect().connect()` when you do this

Reply:
I'm not sure to understand you

Reply:
Do the restart in the 4200 close code, this means we don't need to call both disconnect and reconnect in `.restart()` and `.disconnect()` with a 4200 close code will be handled accordingly